neurodocker/neurodocker.py

```diff
@@ -48,102 +49,140 @@ def _add_generate_common_arguments(parser):
-                       "os_codename (e.g., zesty), server (e.g., usa-nh), and"
-                       " full (if true, use non-free packages). Valid download"
-                       " servers are {}.".format(_ndeb_servers),
-        "spm12": "Install SPM12 and its dependency, Matlab Compiler Runtime."
-                 " Valid keys are version and install_path.",
-        "minc": "Install MINC. Valid keys is version (required), method, and"
-                " install_path. Only version 1.9.15 is supported at this"
-                " time.",
-        "petpvc": "Install PETPVC. Valid keys are version (required), method,"
-                  " and install_path.",
-        "vnc": "Install a VNC server. Valid keys are passwd (required),"
-               " start_at_runtime, and geometry."
+        "all":
+        "Install software packages. Each argument takes a list of"
+        " key=value pairs. Where applicable, the default installation"
+        " behavior is to install by downloading and uncompressing"
+        " binaries. Some programs can be built from source.",
+        "afni":
+        "Install AFNI. Valid keys are version (required), method,"
+        " install_path, install_r, install_r_pkgs, install_python2,"
+        " and install_python3. Only the latest version and version"
+        " 17.2.02 are supported at this time.",
+        "ants":
+        "Install ANTs. Valid keys are version (required), method"
+        " install_path, cmake_opts, and make_opts. Version can be a "
+        " git commit hash if building from source.",
+        "convert3d":
+        "Install Convert3D. Valid keys are version (required),"
+        " method, and install_path.",
+        "dcm2niix":
+        "Install dcm2niix. Valid keys are version, method,"
+        " install_path, cmake_opts, and make_opts",
+        "freesurfer":
+        "Install FreeSurfer. Valid keys are version (required),"
+        " method, install_path, and exclude_paths. A FreeSurfer"
+        " license is required to run the software and is not"
+        " provided by Neurodocker.",
+        "fsl":
+        "Install FSL. Valid keys are version (required), method,"
+        " install_path, and exclude_paths.",
+        "matlabmcr":
+        "Install Matlab Compiler Runtime. Valid keys are version,"
+        " method, and install_path",
+        "miniconda":
+        "Install Miniconda. Valid keys are install_path,"
+        " env_name, conda_install, pip_install, conda_opts,"
+        " pip_opts, activate (default false), and version"
+        " (defaults to latest). The options conda_install and"
+        " pip_install accept strings of packages: conda_install="
+        '"python=3.6 numpy traits".',
+        "mrtrix3":
+        "Install MRtrix3. Valid keys are version (required),"
+        " method, and install_path",
+        "ndfreeze":
+        "Use the NeuroDebian command `nd_freeze` to freeze the apt"
+        " sources to a certain date.",
```

Comment:
again, might be worth adding that of effect only for debian and neurodebian, not ubuntu